// jira.js/out/version2/issues.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Issues = void 0;
const tslib_1 = require("tslib");
class Issues {
    constructor(client) {
        this.client = client;
    }
    getEvents(callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/events',
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createIssue(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/issue',
                method: 'POST',
                params: {
                    updateHistory: parameters === null || parameters === void 0 ? void 0 : parameters.updateHistory,
                },
                data: {
                    transition: parameters === null || parameters === void 0 ? void 0 : parameters.transition,
                    fields: parameters === null || parameters === void 0 ? void 0 : parameters.fields,
                    update: parameters === null || parameters === void 0 ? void 0 : parameters.update,
                    historyMetadata: parameters === null || parameters === void 0 ? void 0 : parameters.historyMetadata,
                    properties: parameters === null || parameters === void 0 ? void 0 : parameters.properties,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createIssues(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/issue/bulk',
                method: 'POST',
                data: {
                    issueUpdates: parameters === null || parameters === void 0 ? void 0 : parameters.issueUpdates,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getCreateIssueMeta(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/issue/createmeta',
                method: 'GET',
                params: {
                    projectIds: parameters === null || parameters === void 0 ? void 0 : parameters.projectIds,
                    projectKeys: parameters === null || parameters === void 0 ? void 0 : parameters.projectKeys,
                    issuetypeIds: parameters === null || parameters === void 0 ? void 0 : parameters.issuetypeIds,
                    issuetypeNames: parameters === null || parameters === void 0 ? void 0 : parameters.issuetypeNames,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getIssue(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}`,
                method: 'GET',
                params: {
                    fields: parameters.fields,
                    fieldsByKeys: parameters.fieldsByKeys,
                    expand: parameters.expand,
                    properties: parameters.properties,
                    updateHistory: parameters.updateHistory,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    editIssue(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}`,
                method: 'PUT',
                params: {
                    notifyUsers: parameters.notifyUsers,
                    overrideScreenSecurity: parameters.overrideScreenSecurity,
                    overrideEditableFlag: parameters.overrideEditableFlag,
                },
                data: {
                    transition: parameters.transition,
                    fields: parameters.fields,
                    update: parameters.update,
                    historyMetadata: parameters.historyMetadata,
                    properties: parameters.properties,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteIssue(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}`,
                method: 'DELETE',
                params: {
                    deleteSubtasks: parameters.deleteSubtasks,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    assignIssue(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/assignee`,
                method: 'PUT',
                data: {
                    self: parameters.self,
                    key: parameters.key,
                    accountId: parameters.accountId,
                    accountType: parameters.accountType,
                    name: parameters.name,
                    emailAddress: parameters.emailAddress,
                    avatarUrls: parameters.avatarUrls,
                    displayName: parameters.displayName,
                    active: parameters.active,
                    timeZone: parameters.timeZone,
                    locale: parameters.locale,
                    groups: parameters.groups,
                    applicationRoles: parameters.applicationRoles,
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getChangeLogs(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/changelog`,
                method: 'GET',
                params: {
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getChangeLogsByIds(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/changelog/list`,
                method: 'POST',
                data: {
                    changelogIds: parameters.changelogIds,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getEditIssueMeta(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/editmeta`,
                method: 'GET',
                params: {
                    overrideScreenSecurity: parameters.overrideScreenSecurity,
                    overrideEditableFlag: parameters.overrideEditableFlag,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    notify(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/notify`,
                method: 'POST',
                data: {
                    subject: parameters.subject,
                    textBody: parameters.textBody,
                    htmlBody: parameters.htmlBody,
                    to: parameters.to,
                    restrict: parameters.restrict,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getTransitions(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/transitions`,
                method: 'GET',
                params: {
                    expand: parameters.expand,
                    transitionId: parameters.transitionId,
                    skipRemoteOnlyCondition: parameters.skipRemoteOnlyCondition,
                    includeUnavailableTransitions: parameters.includeUnavailableTransitions,
                    sortByOpsBarAndStatus: parameters.sortByOpsBarAndStatus,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    doTransition(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/issue/${parameters.issueIdOrKey}/transitions`,
                method: 'POST',
                data: {
                    transition: parameters.transition,
                    fields: parameters.fields,
                    update: parameters.update,
                    historyMetadata: parameters.historyMetadata,
                    properties: parameters.properties,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Issues = Issues;
