// jira.js/out/version2/issueCustomFieldConfigurationApps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IssueCustomFieldConfigurationApps = void 0;
const tslib_1 = require("tslib");
class IssueCustomFieldConfigurationApps {
    constructor(client) {
        this.client = client;
    }
    getCustomFieldConfiguration(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/app/field/${parameters.fieldIdOrKey}/context/configuration`,
                method: 'GET',
                params: {
                    id: parameters.id,
                    contextId: parameters.contextId,
                    fieldContextId: parameters.fieldContextId,
                    issueId: parameters.issueId,
                    projectKeyOrId: parameters.projectKeyOrId,
                    issueTypeId: parameters.issueTypeId,
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateCustomFieldConfiguration(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/app/field/${parameters.fieldIdOrKey}/context/configuration`,
                method: 'PUT',
                data: {
                    configurations: parameters.configurations,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.IssueCustomFieldConfigurationApps = IssueCustomFieldConfigurationApps;
