// jira.js/out/version2/jiraExpressions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JiraExpressions = void 0;
const tslib_1 = require("tslib");
class JiraExpressions {
    constructor(client) {
        this.client = client;
    }
    analyseExpression(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/expression/analyse',
                method: 'POST',
                params: {
                    check: parameters === null || parameters === void 0 ? void 0 : parameters.check,
                },
                data: {
                    expressions: parameters === null || parameters === void 0 ? void 0 : parameters.expressions,
                    contextVariables: parameters === null || parameters === void 0 ? void 0 : parameters.contextVariables,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    evaluateJiraExpression(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/expression/eval',
                method: 'POST',
                params: {
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                },
                data: {
                    expression: parameters === null || parameters === void 0 ? void 0 : parameters.expression,
                    context: parameters === null || parameters === void 0 ? void 0 : parameters.context,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.JiraExpressions = JiraExpressions;
