// jira.js/out/version2/projectVersions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectVersions = void 0;
const tslib_1 = require("tslib");
class ProjectVersions {
    constructor(client) {
        this.client = client;
    }
    getProjectVersionsPaginated(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/project/${parameters.projectIdOrKey}/version`,
                method: 'GET',
                params: {
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                    orderBy: parameters.orderBy,
                    query: parameters.query,
                    status: parameters.status,
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProjectVersions(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/project/${parameters.projectIdOrKey}/versions`,
                method: 'GET',
                params: {
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createVersion(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/version',
                method: 'POST',
                data: {
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    self: parameters === null || parameters === void 0 ? void 0 : parameters.self,
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    archived: parameters === null || parameters === void 0 ? void 0 : parameters.archived,
                    released: parameters === null || parameters === void 0 ? void 0 : parameters.released,
                    startDate: parameters === null || parameters === void 0 ? void 0 : parameters.startDate,
                    releaseDate: parameters === null || parameters === void 0 ? void 0 : parameters.releaseDate,
                    overdue: parameters === null || parameters === void 0 ? void 0 : parameters.overdue,
                    userStartDate: parameters === null || parameters === void 0 ? void 0 : parameters.userStartDate,
                    userReleaseDate: parameters === null || parameters === void 0 ? void 0 : parameters.userReleaseDate,
                    project: parameters === null || parameters === void 0 ? void 0 : parameters.project,
                    projectId: parameters === null || parameters === void 0 ? void 0 : parameters.projectId,
                    moveUnfixedIssuesTo: parameters === null || parameters === void 0 ? void 0 : parameters.moveUnfixedIssuesTo,
                    operations: parameters === null || parameters === void 0 ? void 0 : parameters.operations,
                    issuesStatusForFixVersion: parameters === null || parameters === void 0 ? void 0 : parameters.issuesStatusForFixVersion,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getVersion(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}`,
                method: 'GET',
                params: {
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateVersion(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}`,
                method: 'PUT',
                data: {
                    expand: parameters.expand,
                    description: parameters.description,
                    name: parameters.name,
                    archived: parameters.archived,
                    released: parameters.released,
                    startDate: parameters.startDate,
                    releaseDate: parameters.releaseDate,
                    project: parameters.project,
                    projectId: parameters.projectId,
                    moveUnfixedIssuesTo: parameters.moveUnfixedIssuesTo,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteVersion(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}`,
                method: 'DELETE',
                params: {
                    moveFixIssuesTo: parameters.moveFixIssuesTo,
                    moveAffectedIssuesTo: parameters.moveAffectedIssuesTo,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    mergeVersions(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}/mergeto/${parameters.moveIssuesTo}`,
                method: 'PUT',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    moveVersion(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}/move`,
                method: 'POST',
                data: {
                    after: parameters.after,
                    position: parameters.position,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getVersionRelatedIssues(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}/relatedIssueCounts`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteAndReplaceVersion(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}/removeAndSwap`,
                method: 'POST',
                data: {
                    moveFixIssuesTo: parameters.moveFixIssuesTo,
                    moveAffectedIssuesTo: parameters.moveAffectedIssuesTo,
                    customFieldReplacementList: parameters.customFieldReplacementList,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getVersionUnresolvedIssues(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/version/${parameters.id}/unresolvedIssueCount`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.ProjectVersions = ProjectVersions;
