// jira.js/out/version2/projectRoles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectRoles = void 0;
const tslib_1 = require("tslib");
class ProjectRoles {
    constructor(client) {
        this.client = client;
    }
    getProjectRoles(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/project/${parameters.projectIdOrKey}/role`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProjectRole(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/project/${parameters.projectIdOrKey}/role/${parameters.id}`,
                method: 'GET',
                params: {
                    excludeInactiveUsers: parameters.excludeInactiveUsers,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProjectRoleDetails(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/project/${parameters.projectIdOrKey}/roledetails`,
                method: 'GET',
                params: {
                    currentMember: parameters.currentMember,
                    excludeConnectAddons: parameters.excludeConnectAddons,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getAllProjectRoles(callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/role',
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createProjectRole(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/role',
                method: 'POST',
                data: {
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProjectRoleById(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/role/${parameters.id}`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    partialUpdateProjectRole(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/role/${parameters.id}`,
                method: 'POST',
                data: {
                    name: parameters.name,
                    description: parameters.description,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    fullyUpdateProjectRole(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/role/${parameters.id}`,
                method: 'PUT',
                data: {
                    name: parameters.name,
                    description: parameters.description,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteProjectRole(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/2/role/${parameters.id}`,
                method: 'DELETE',
                params: {
                    swap: parameters.swap,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.ProjectRoles = ProjectRoles;
