// jira.js/out/version2/users.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Users = void 0;
const tslib_1 = require("tslib");
const paramSerializer_1 = require("../paramSerializer");
class Users {
    constructor(client) {
        this.client = client;
    }
    getUser(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user',
                method: 'GET',
                params: {
                    accountId: parameters === null || parameters === void 0 ? void 0 : parameters.accountId,
                    username: parameters === null || parameters === void 0 ? void 0 : parameters.username,
                    key: parameters === null || parameters === void 0 ? void 0 : parameters.key,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createUser(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user',
                method: 'POST',
                data: {
                    self: parameters === null || parameters === void 0 ? void 0 : parameters.self,
                    key: parameters === null || parameters === void 0 ? void 0 : parameters.key,
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    password: parameters === null || parameters === void 0 ? void 0 : parameters.password,
                    emailAddress: parameters === null || parameters === void 0 ? void 0 : parameters.emailAddress,
                    displayName: parameters === null || parameters === void 0 ? void 0 : parameters.displayName,
                    applicationKeys: parameters === null || parameters === void 0 ? void 0 : parameters.applicationKeys,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    removeUser(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user',
                method: 'DELETE',
                params: {
                    accountId: parameters.accountId,
                    username: parameters.username,
                    key: parameters.key,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    bulkGetUsers(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/bulk',
                method: 'GET',
                params: {
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                    username: parameters.username,
                    key: parameters.key,
                    accountId: (0, paramSerializer_1.paramSerializer)('accountId', parameters.accountId),
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    bulkGetUsersMigration(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/bulk/migration',
                method: 'GET',
                params: {
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                    username: (0, paramSerializer_1.paramSerializer)('username', parameters.username),
                    key: (0, paramSerializer_1.paramSerializer)('key', parameters.key),
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getUserDefaultColumns(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/columns',
                method: 'GET',
                params: {
                    accountId: parameters === null || parameters === void 0 ? void 0 : parameters.accountId,
                    username: parameters === null || parameters === void 0 ? void 0 : parameters.username,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    setUserColumns(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/columns',
                method: 'PUT',
                params: {
                    accountId: parameters === null || parameters === void 0 ? void 0 : parameters.accountId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    resetUserColumns(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/columns',
                method: 'DELETE',
                params: {
                    accountId: parameters === null || parameters === void 0 ? void 0 : parameters.accountId,
                    username: parameters === null || parameters === void 0 ? void 0 : parameters.username,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getUserEmail(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/email',
                method: 'GET',
                params: {
                    accountId: parameters.accountId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getUserEmailBulk(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/email/bulk',
                method: 'GET',
                params: {
                    accountId: parameters.accountId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getUserGroups(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/user/groups',
                method: 'GET',
                params: {
                    accountId: parameters.accountId,
                    username: parameters.username,
                    key: parameters.key,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getAllUsersDefault(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/users',
                method: 'GET',
                params: {
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getAllUsers(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/2/users/search',
                method: 'GET',
                params: {
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Users = Users;
