// jira.js/out/agile/backlog.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Backlog = void 0;
const tslib_1 = require("tslib");
class Backlog {
    constructor(client) {
        this.client = client;
    }
    moveIssuesToBacklog(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/agile/1.0/backlog/issue',
                method: 'POST',
                data: {
                    issues: parameters === null || parameters === void 0 ? void 0 : parameters.issues,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    moveIssuesToBacklogForBoard(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/backlog/${parameters.boardId}/issue`,
                method: 'POST',
                data: {
                    issues: parameters.issues,
                    rankBeforeIssue: parameters.rankBeforeIssue,
                    rankAfterIssue: parameters.rankAfterIssue,
                    rankCustomFieldId: parameters.rankCustomFieldId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Backlog = Backlog;
