// jira.js/out/agile/issue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Issue = void 0;
const tslib_1 = require("tslib");
class Issue {
    constructor(client) {
        this.client = client;
    }
    rankIssues(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/agile/1.0/issue/rank',
                method: 'PUT',
                data: {
                    issues: parameters === null || parameters === void 0 ? void 0 : parameters.issues,
                    rankBeforeIssue: parameters === null || parameters === void 0 ? void 0 : parameters.rankBeforeIssue,
                    rankAfterIssue: parameters === null || parameters === void 0 ? void 0 : parameters.rankAfterIssue,
                    rankCustomFieldId: parameters === null || parameters === void 0 ? void 0 : parameters.rankCustomFieldId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getIssue(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/issue/${parameters.issueIdOrKey}`,
                method: 'GET',
                params: {
                    fields: parameters.fields,
                    expand: parameters.expand,
                    updateHistory: parameters.updateHistory,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getIssueEstimationForBoard(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/issue/${parameters.issueIdOrKey}/estimation`,
                method: 'GET',
                params: {
                    boardId: parameters.boardId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    estimateIssueForBoard(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/issue/${parameters.issueIdOrKey}/estimation`,
                method: 'PUT',
                params: {
                    boardId: parameters.boardId,
                },
                data: {
                    value: parameters.value,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Issue = Issue;
