// jira.js/out/agile/sprint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sprint = void 0;
const tslib_1 = require("tslib");
class Sprint {
    constructor(client) {
        this.client = client;
    }
    createSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/agile/1.0/sprint',
                method: 'POST',
                data: {
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    startDate: parameters === null || parameters === void 0 ? void 0 : parameters.startDate,
                    endDate: parameters === null || parameters === void 0 ? void 0 : parameters.endDate,
                    originBoardId: parameters === null || parameters === void 0 ? void 0 : parameters.originBoardId,
                    goal: parameters === null || parameters === void 0 ? void 0 : parameters.goal,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    partiallyUpdateSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
                method: 'POST',
                data: {
                    id: parameters.id,
                    self: parameters.self,
                    state: parameters.state,
                    name: parameters.name,
                    startDate: parameters.startDate,
                    endDate: parameters.endDate,
                    completeDate: parameters.completeDate,
                    createdDate: parameters.createdDate,
                    originBoardId: parameters.originBoardId,
                    goal: parameters.goal,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
                method: 'PUT',
                data: {
                    id: parameters.id,
                    self: parameters.self,
                    state: parameters.state,
                    name: parameters.name,
                    startDate: parameters.startDate,
                    endDate: parameters.endDate,
                    completeDate: parameters.completeDate,
                    createdDate: parameters.createdDate,
                    originBoardId: parameters.originBoardId,
                    goal: parameters.goal,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getIssuesForSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/issue`,
                method: 'GET',
                params: {
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                    jql: parameters.jql,
                    validateQuery: parameters.validateQuery,
                    fields: parameters.fields,
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    moveIssuesToSprintAndRank(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/issue`,
                method: 'POST',
                data: {
                    issues: parameters.issues,
                    rankBeforeIssue: parameters.rankBeforeIssue,
                    rankAfterIssue: parameters.rankAfterIssue,
                    rankCustomFieldId: parameters.rankCustomFieldId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getPropertiesKeys(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties/${parameters.propertyKey}`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    setProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties/${parameters.propertyKey}`,
                method: 'PUT',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteProperty(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/properties/${parameters.propertyKey}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    swapSprint(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/agile/1.0/sprint/${parameters.sprintId}/swap`,
                method: 'POST',
                data: {
                    sprintToSwapWith: parameters.sprintToSwapWith,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Sprint = Sprint;
