// jira.js/out/version3/filters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Filters = void 0;
const tslib_1 = require("tslib");
class Filters {
    constructor(client) {
        this.client = client;
    }
    getFilters(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/filter',
                method: 'GET',
                params: {
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createFilter(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/filter',
                method: 'POST',
                params: {
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    overrideSharePermissions: parameters === null || parameters === void 0 ? void 0 : parameters.overrideSharePermissions,
                },
                data: {
                    self: parameters === null || parameters === void 0 ? void 0 : parameters.self,
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                    owner: parameters === null || parameters === void 0 ? void 0 : parameters.owner,
                    jql: parameters === null || parameters === void 0 ? void 0 : parameters.jql,
                    viewUrl: parameters === null || parameters === void 0 ? void 0 : parameters.viewUrl,
                    searchUrl: parameters === null || parameters === void 0 ? void 0 : parameters.searchUrl,
                    favourite: parameters === null || parameters === void 0 ? void 0 : parameters.favourite,
                    favouritedCount: parameters === null || parameters === void 0 ? void 0 : parameters.favouritedCount,
                    sharePermissions: parameters === null || parameters === void 0 ? void 0 : parameters.sharePermissions,
                    editPermissions: parameters === null || parameters === void 0 ? void 0 : parameters.editPermissions,
                    sharedUsers: parameters === null || parameters === void 0 ? void 0 : parameters.sharedUsers,
                    subscriptions: parameters === null || parameters === void 0 ? void 0 : parameters.subscriptions,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getFavouriteFilters(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/filter/favourite',
                method: 'GET',
                params: {
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getMyFilters(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/filter/my',
                method: 'GET',
                params: {
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    includeFavourites: parameters === null || parameters === void 0 ? void 0 : parameters.includeFavourites,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getFiltersPaginated(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/filter/search',
                method: 'GET',
                params: {
                    filterName: parameters === null || parameters === void 0 ? void 0 : parameters.filterName,
                    accountId: parameters === null || parameters === void 0 ? void 0 : parameters.accountId,
                    owner: parameters === null || parameters === void 0 ? void 0 : parameters.owner,
                    groupname: parameters === null || parameters === void 0 ? void 0 : parameters.groupname,
                    groupId: parameters === null || parameters === void 0 ? void 0 : parameters.groupId,
                    projectId: parameters === null || parameters === void 0 ? void 0 : parameters.projectId,
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                    orderBy: parameters === null || parameters === void 0 ? void 0 : parameters.orderBy,
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    overrideSharePermissions: parameters === null || parameters === void 0 ? void 0 : parameters.overrideSharePermissions,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getFilter(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}`,
                method: 'GET',
                params: {
                    expand: parameters.expand,
                    overrideSharePermissions: parameters.overrideSharePermissions,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateFilter(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}`,
                method: 'PUT',
                params: {
                    expand: parameters.expand,
                    overrideSharePermissions: parameters.overrideSharePermissions,
                },
                data: {
                    name: parameters.name,
                    description: parameters.description,
                    jql: parameters.jql,
                    favourite: parameters.favourite,
                    sharePermissions: parameters.sharePermissions,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteFilter(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getColumns(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}/columns`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    setColumns(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}/columns`,
                method: 'PUT',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    resetColumns(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}/columns`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    setFavouriteForFilter(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}/favourite`,
                method: 'PUT',
                params: {
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteFavouriteForFilter(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}/favourite`,
                method: 'DELETE',
                params: {
                    expand: parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    changeFilterOwner(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/filter/${parameters.id}/owner`,
                method: 'PUT',
                data: {
                    accountId: parameters.accountId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.Filters = Filters;
