// jira.js/out/version3/issueSearch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IssueSearch = void 0;
const tslib_1 = require("tslib");
class IssueSearch {
    constructor(client) {
        this.client = client;
    }
    getIssuePickerResource(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/issue/picker',
                method: 'GET',
                params: {
                    query: parameters === null || parameters === void 0 ? void 0 : parameters.query,
                    currentJQL: parameters === null || parameters === void 0 ? void 0 : parameters.currentJQL,
                    currentIssueKey: parameters === null || parameters === void 0 ? void 0 : parameters.currentIssueKey,
                    currentProjectId: parameters === null || parameters === void 0 ? void 0 : parameters.currentProjectId,
                    showSubTasks: parameters === null || parameters === void 0 ? void 0 : parameters.showSubTasks,
                    showSubTaskParent: parameters === null || parameters === void 0 ? void 0 : parameters.showSubTaskParent,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    matchIssues(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/jql/match',
                method: 'POST',
                data: {
                    jqls: parameters === null || parameters === void 0 ? void 0 : parameters.jqls,
                    issueIds: parameters === null || parameters === void 0 ? void 0 : parameters.issueIds,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    searchForIssuesUsingJql(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/search',
                method: 'GET',
                params: {
                    jql: parameters === null || parameters === void 0 ? void 0 : parameters.jql,
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                    validateQuery: parameters === null || parameters === void 0 ? void 0 : parameters.validateQuery,
                    fields: parameters === null || parameters === void 0 ? void 0 : parameters.fields,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    properties: parameters === null || parameters === void 0 ? void 0 : parameters.properties,
                    fieldsByKeys: parameters === null || parameters === void 0 ? void 0 : parameters.fieldsByKeys,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    searchForIssuesUsingJqlPost(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/search',
                method: 'POST',
                data: {
                    jql: parameters === null || parameters === void 0 ? void 0 : parameters.jql,
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                    fields: parameters === null || parameters === void 0 ? void 0 : parameters.fields,
                    validateQuery: parameters === null || parameters === void 0 ? void 0 : parameters.validateQuery,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                    properties: parameters === null || parameters === void 0 ? void 0 : parameters.properties,
                    fieldsByKeys: parameters === null || parameters === void 0 ? void 0 : parameters.fieldsByKeys,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.IssueSearch = IssueSearch;
