// jira.js/out/version3/issuePriorities.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IssuePriorities = void 0;
const tslib_1 = require("tslib");
class IssuePriorities {
    constructor(client) {
        this.client = client;
    }
    getPriorities(callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/priority',
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createPriority(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/priority',
                method: 'POST',
                data: {
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                    iconUrl: parameters === null || parameters === void 0 ? void 0 : parameters.iconUrl,
                    statusColor: parameters === null || parameters === void 0 ? void 0 : parameters.statusColor,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    setDefaultPriority(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/priority/default',
                method: 'PUT',
                data: {
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    searchPriorities(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/priority/search',
                method: 'GET',
                params: {
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                    onlyDefault: parameters === null || parameters === void 0 ? void 0 : parameters.onlyDefault,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getPriority(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/priority/${parameters.id}`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updatePriority(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/priority/${parameters.id}`,
                method: 'PUT',
                data: {
                    name: parameters.name,
                    description: parameters.description,
                    iconUrl: parameters.iconUrl,
                    statusColor: parameters.statusColor,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.IssuePriorities = IssuePriorities;
