// jira.js/out/version3/projectComponents.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectComponents = void 0;
const tslib_1 = require("tslib");
class ProjectComponents {
    constructor(client) {
        this.client = client;
    }
    createComponent(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/component',
                method: 'POST',
                data: {
                    self: parameters === null || parameters === void 0 ? void 0 : parameters.self,
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                    lead: parameters === null || parameters === void 0 ? void 0 : parameters.lead,
                    leadUserName: parameters === null || parameters === void 0 ? void 0 : parameters.leadUserName,
                    leadAccountId: parameters === null || parameters === void 0 ? void 0 : parameters.leadAccountId,
                    assigneeType: parameters === null || parameters === void 0 ? void 0 : parameters.assigneeType,
                    assignee: parameters === null || parameters === void 0 ? void 0 : parameters.assignee,
                    realAssigneeType: parameters === null || parameters === void 0 ? void 0 : parameters.realAssigneeType,
                    realAssignee: parameters === null || parameters === void 0 ? void 0 : parameters.realAssignee,
                    isAssigneeTypeValid: parameters === null || parameters === void 0 ? void 0 : parameters.isAssigneeTypeValid,
                    project: parameters === null || parameters === void 0 ? void 0 : parameters.project,
                    projectId: parameters === null || parameters === void 0 ? void 0 : parameters.projectId,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getComponent(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/component/${parameters.id}`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateComponent(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/component/${parameters.id}`,
                method: 'PUT',
                data: {
                    name: parameters.name,
                    description: parameters.description,
                    leadUserName: parameters.leadUserName,
                    leadAccountId: parameters.leadAccountId,
                    assigneeType: parameters.assigneeType,
                    project: parameters.project,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteComponent(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/component/${parameters.id}`,
                method: 'DELETE',
                params: {
                    moveIssuesTo: parameters.moveIssuesTo,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getComponentRelatedIssues(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/component/${parameters.id}/relatedIssueCounts`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProjectComponentsPaginated(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/project/${parameters.projectIdOrKey}/component`,
                method: 'GET',
                params: {
                    startAt: parameters.startAt,
                    maxResults: parameters.maxResults,
                    orderBy: parameters.orderBy,
                    query: parameters.query,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getProjectComponents(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/project/${parameters.projectIdOrKey}/components`,
                method: 'GET',
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.ProjectComponents = ProjectComponents;
