// jira.js/out/version3/uIModificationsApps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UIModificationsApps = void 0;
const tslib_1 = require("tslib");
class UIModificationsApps {
    constructor(client) {
        this.client = client;
    }
    getUiModifications(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/uiModifications',
                method: 'GET',
                params: {
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                    expand: parameters === null || parameters === void 0 ? void 0 : parameters.expand,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createUiModification(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/uiModifications',
                method: 'POST',
                data: {
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                    data: parameters === null || parameters === void 0 ? void 0 : parameters.data,
                    contexts: parameters === null || parameters === void 0 ? void 0 : parameters.contexts,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateUiModification(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/uiModifications/${parameters.uiModificationId}`,
                method: 'PUT',
                data: {
                    name: parameters.name,
                    description: parameters.description,
                    data: parameters.data,
                    contexts: parameters.contexts,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteUiModification(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/uiModifications/${parameters.uiModificationId}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.UIModificationsApps = UIModificationsApps;
