// jira.js/out/version3/workflowSchemes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowSchemes = void 0;
const tslib_1 = require("tslib");
class WorkflowSchemes {
    constructor(client) {
        this.client = client;
    }
    getAllWorkflowSchemes(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/workflowscheme',
                method: 'GET',
                params: {
                    startAt: parameters === null || parameters === void 0 ? void 0 : parameters.startAt,
                    maxResults: parameters === null || parameters === void 0 ? void 0 : parameters.maxResults,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    createWorkflowScheme(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: '/rest/api/3/workflowscheme',
                method: 'POST',
                data: {
                    id: parameters === null || parameters === void 0 ? void 0 : parameters.id,
                    name: parameters === null || parameters === void 0 ? void 0 : parameters.name,
                    description: parameters === null || parameters === void 0 ? void 0 : parameters.description,
                    defaultWorkflow: parameters === null || parameters === void 0 ? void 0 : parameters.defaultWorkflow,
                    issueTypeMappings: parameters === null || parameters === void 0 ? void 0 : parameters.issueTypeMappings,
                    originalDefaultWorkflow: parameters === null || parameters === void 0 ? void 0 : parameters.originalDefaultWorkflow,
                    originalIssueTypeMappings: parameters === null || parameters === void 0 ? void 0 : parameters.originalIssueTypeMappings,
                    draft: parameters === null || parameters === void 0 ? void 0 : parameters.draft,
                    lastModifiedUser: parameters === null || parameters === void 0 ? void 0 : parameters.lastModifiedUser,
                    lastModified: parameters === null || parameters === void 0 ? void 0 : parameters.lastModified,
                    self: parameters === null || parameters === void 0 ? void 0 : parameters.self,
                    updateDraftIfNeeded: parameters === null || parameters === void 0 ? void 0 : parameters.updateDraftIfNeeded,
                    issueTypes: parameters === null || parameters === void 0 ? void 0 : parameters.issueTypes,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getWorkflowScheme(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}`,
                method: 'GET',
                params: {
                    returnDraftIfExists: parameters.returnDraftIfExists,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateWorkflowScheme(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}`,
                method: 'PUT',
                data: {
                    name: parameters.name,
                    description: parameters.description,
                    defaultWorkflow: parameters.defaultWorkflow,
                    issueTypeMappings: parameters.issueTypeMappings,
                    updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteWorkflowScheme(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}`,
                method: 'DELETE',
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getDefaultWorkflow(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/default`,
                method: 'GET',
                params: {
                    returnDraftIfExists: parameters.returnDraftIfExists,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateDefaultWorkflow(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/default`,
                method: 'PUT',
                data: {
                    workflow: parameters.workflow,
                    updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteDefaultWorkflow(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/default`,
                method: 'DELETE',
                params: {
                    updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getWorkflowSchemeIssueType(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/issuetype/${parameters.issueType}`,
                method: 'GET',
                params: {
                    returnDraftIfExists: parameters.returnDraftIfExists,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    setWorkflowSchemeIssueType(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/issuetype/${parameters.issueType}`,
                method: 'PUT',
                data: parameters.body,
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteWorkflowSchemeIssueType(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/issuetype/${parameters.issueType}`,
                method: 'DELETE',
                params: {
                    updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    getWorkflow(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/workflow`,
                method: 'GET',
                params: {
                    workflowName: parameters.workflowName,
                    returnDraftIfExists: parameters.returnDraftIfExists,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    updateWorkflowMapping(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/workflow`,
                method: 'PUT',
                params: {
                    workflowName: parameters.workflowName,
                },
                data: {
                    workflow: parameters.workflow,
                    issueTypes: parameters.issueTypes,
                    defaultMapping: parameters.defaultMapping,
                    updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
    deleteWorkflowMapping(parameters, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const config = {
                url: `/rest/api/3/workflowscheme/${parameters.id}/workflow`,
                method: 'DELETE',
                params: {
                    workflowName: parameters.workflowName,
                    updateDraftIfNeeded: parameters.updateDraftIfNeeded,
                },
            };
            return this.client.sendRequest(config, callback);
        });
    }
}
exports.WorkflowSchemes = WorkflowSchemes;
