// jira.js/out/version3/parameters/getRecent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRecent = void 0;
var GetRecent;
(function (GetRecent) {
    let Expand;
    (function (Expand) {
        /** Returns the project description. */
        Expand["Description"] = "description";
        /** Returns all project keys associated with a project. */
        Expand["ProjectKeys"] = "projectKeys";
        /** Returns information about the project lead. */
        Expand["Lead"] = "lead";
        /** Returns all issue types associated with the project. */
        Expand["IssueTypes"] = "issueTypes";
        /** Returns the URL associated with the project. */
        Expand["Url"] = "url";
        /** Returns the permissions associated with the project. */
        Expand["Permissions"] = "permissions";
        /** EXPERIMENTAL. Returns the insight details of total issue count and last issue update time for the project. */
        Expand["Insight"] = "insight";
        /** Returns the project with all available expand options. */
        Expand["All"] = "*";
    })(Expand = GetRecent.Expand || (GetRecent.Expand = {}));
})(GetRecent = exports.GetRecent || (exports.GetRecent = {}));
