// jira.js/out/version2/parameters/getFiltersPaginated.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetFiltersPaginated = void 0;
var GetFiltersPaginated;
(function (GetFiltersPaginated) {
    let Expand;
    (function (Expand) {
        /** Returns the description of the filter. */
        Expand["Description"] = "description";
        /** Returns an indicator of whether the user has set the filter as a favorite. */
        Expand["Favourite"] = "favourite";
        /** Returns a count of how many users have set this filter as a favorite. */
        Expand["FavouritedCount"] = "favouritedCount";
        /** Returns the JQL query that the filter uses. */
        Expand["JQL"] = "jql";
        /** Returns the owner of the filter. */
        Expand["Owner"] = "owner";
        /** Returns a URL to perform the filter's JQL query. */
        Expand["SearchUrl"] = "searchUrl";
        /** Returns the share permissions defined for the filter. */
        Expand["SharePermissions"] = "sharePermissions";
        /** Returns the edit permissions defined for the filter. */
        Expand["EditPermissions"] = "editPermissions";
        /** Returns whether the current user has permission to edit the filter. */
        Expand["IsWritable"] = "isWritable";
        /** Returns the users that are subscribed to the filter. */
        Expand["Subscriptions"] = "subscriptions";
        /** Returns a URL to view the filter. */
        Expand["ViewUrl"] = "viewUrl";
    })(Expand = GetFiltersPaginated.Expand || (GetFiltersPaginated.Expand = {}));
})(GetFiltersPaginated = exports.GetFiltersPaginated || (exports.GetFiltersPaginated = {}));
